core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -3280,22 +3280,21 @@ private CompiledQuery executorSelector(SqlExecutionContext executionContext) thr
                     if (tok == null || Chars.equals(tok, ';')) {
                         return dropTable(executionContext, tableName, tableNamePosition, hasIfExists);
                     }
-                    throw parseErrorUnexpected("[;]", tok);
+                    return unknownDropTableSuffix(executionContext, tok, tableName, tableNamePosition, hasIfExists);
                 }
-
                 // DROP ALL TABLES [;]
-                if (SqlKeywords.isAllKeyword(tok)) {
+                else if (SqlKeywords.isAllKeyword(tok)) {
```

Comment:
nit: `else` is redundant here
```suggestion
                if (SqlKeywords.isAllKeyword(tok)) {
```